docs/concepts/resource-manager.md

```diff
@@ -683,6 +683,38 @@ Both is required to ensure system component `Pod`s can be _scheduled_ or _execut
 
 > You can opt-out of this behaviour for `Pod`s by labeling them with `system-components-config.resources.gardener.cloud/skip=true`.
 
+#### EndpointSlice Hints
+
+This webhook mutates [`EndpointSlice`s](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/). For each endpoint in the EndpointSlice, it sets the endpoint's hints to the endpoint's zone.
+
+```yaml
+apiVersion: discovery.k8s.io/v1
+kind: EndpointSlice
+metadata:
+  name: example-hints
+endpoints:
+- addresses:
+  - "10.1.2.3"
+  conditions:
+    ready: true
+  hostname: pod-1
+  zone: zone-a
+  hints:
+    forZones:
+    - name: "zone-a" # added by webhook
+- addresses:
+  - "10.1.2.4"
+  conditions:
+    ready: true
+  hostname: pod-2
+  zone: zone-b
+  hints:
+    forZones:
+    - name: "zone-b" # added by webhook
+```
+
+The webhook aims to circumvent issues with the Kubernetes `TopologyAwareHints` feature that currently does not allow to achieve a deterministic topology-aware traffic routing. For more details, see the following issue [kubernetes/kubernetes#113731](https://github.com/kubernetes/kubernetes/issues/113731) that describes drawbacks of the `TopologyAwareHints` feature for our use case.
+
```

Comment:
You could add a link to https://github.com/kubernetes/enhancements/pull/3765 and explain that the webhook will be dropped as soon as this feature is natively supported.

Reply:
I added a general sentence that the webhook can be potentially dropped if/when there is native support for deterministic topology-aware traffic routing in Kubernetes. I will avoid putting a link to KEP as there are already several KEPs discussing or trying to address this. And discussions in them are not leading to somewhere for now. If someone is interested in details, can check the information in https://github.com/kubernetes/enhancements/pull/3765.